velox/docs/develop/operators.rst

```diff
@@ -291,6 +292,82 @@ Note: Presto allows multiple global grouping sets in a single SQL query.
 
 Hence, globalGroupingSets is a vector of groupIds.
 
+.. _ExpandNode:
+
+ExpandNode
+~~~~~~~~~~~
+
+For each input row, generates N rows with M columns according to specified 'projections'.
+'projections' is an N x M matrix of expressions: a vector of N rows each having M columns.
+Each expression is either a column reference or a constant. Both null and non-null constants are allowed.
+'names' is a list of M new column names. The semantic of this operator matches Spark. Using project and unnest can be
+employed to implement the expand functionality. However, the performance is suboptimal when creating an array
+constructor within the Project operation.
+
+.. list-table::
+   :widths: 10 30
+   :align: left
+   :header-rows: 1
+
+   * - Property
+     - Description
+   * - projections
+     - A vector of N rows each having M columns. Each expression is either a column reference or a constant.
+   * - names
+     - A list of new column names.
+
+ExpandNode is typically used to compute GROUPING SETS, CUBE, ROLLUP and COUNT DISTINCT.   
+
+To illustrate how ExpandNode works so lets examine the following SQL query:
+
+.. code-block:: sql
+
+  Select sum(l_suppkey) from lineitem where group by ROLLUP(l_orderkey, l_partkey);
+
+In the planning phase, Spark generates an Expand operator with the following projection list:
+
+.. code-block::
+
+  [l_suppkey, l_orderkey, l_partkey, 0],
+  [l_suppkey, l_orderkey, null,      1],
+  [l_suppkey, null,       null,      3]
```

Comment:
Curious, why 3? It looks like a typo as a reader would expect 2. If 3 is correct, perhaps, add a short explanation to avoid confusing future readers.

Reply:
@mbasmanova Add the document to explain why it is 3 not 2. Please help to review. Thanks.